test/service_manager/service_manager.go

```diff
@@ -0,0 +1,166 @@
+}
+
+const (
+	canaryTestTopicName      = "__strimzi_canary_test3"
+	kafkaBrokerAddress       = "127.0.0.1:9092"
+	canaryRetentionTime      = "1000"
+
+	pathToDockerComposeImage = "test/compose-kafka-zookeeper.yaml"
+	pathToMainMethod         = "cmd/main.go"
+)
+
+func (c *ServiceManager) StartKafkaZookeeperContainers() {
+	log.Println("Starting kafka & Zookeeper")
+
+	var cmd = exec.Command("docker-compose", "-f", c.pathDockerComposeKafkaZookeeper, "up", "-d" )
+	if err := cmd.Run(); err != nil {
+		log.Fatal(err)
+	}
+	log.Println("Zookeeper and Kafka containers created")
+	// after creation of containers we still have to wait for some time before successful communication with Kafka & Zookeper
+	c.waitForBroker()
+}
+
+func (c *ServiceManager) StopKafkaZookeeperContainers() {
+	var cmd = exec.Command("docker-compose", "-f", c.pathDockerComposeKafkaZookeeper, "down")
+	if err := cmd.Start(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func CreateManager() *ServiceManager {
+	manager := &ServiceManager{}
+
+	manager.RetentionTime = canaryRetentionTime
+	manager.TopicTestName = canaryTestTopicName
+	manager.pathToCanaryMain = pathToMainMethod
+	manager.pathDockerComposeKafkaZookeeper = pathToDockerComposeImage
+	manager.KafkaBrokerAddress = kafkaBrokerAddress
+	return manager
+}
+
+func (c *ServiceManager) StartCanary() {
+	log.Println("Starting Canary")
+	c.setUpCanaryParamsViaEnv()
+	myCmd := exec.Command("go", "run",  c.pathToCanaryMain )
+	//myCmd.Stdout = os.Stdout
+	//myCmd.Stderr = os.Stderr
+
+	if err := myCmd.Start(); err != nil {
+		log.Fatal(err.Error())
+	}
+	// before we start to consider canary prepared we wait for it to create expected topic
+	c.waitForCanarySetUp()
+
+}
+
+// per se it means waiting for container's broker to communicate correctly
+func (c *ServiceManager) waitForBroker(){
+	log.Println("start waiting for broker")
+	timeout := time.After(10 * time.Second)
+	brokerIsReadyChannel := make(chan bool)
+
+	go func() {
+		configuration := sarama.NewConfig()
+		brokers := []string{c.KafkaBrokerAddress}
+
+		for ;; {
+			consumer, err := sarama.NewConsumer(brokers, configuration)
+			if err != nil {
+				log.Println("waiting for broker's start")
+				time.Sleep(time.Millisecond * 500)
+				continue
+			}
+			// if we can create consumer broker can communicate
+			consumer.Close()
+			break
+		}
+
+		brokerIsReadyChannel <- true
+	}()
+
+	select {
+	case <-timeout:
+		log.Fatal("Broker isn't ready within expected timeout")
+	case <-brokerIsReadyChannel:
+		log.Println("Container (Broker) is ready")
+	}
+}
+
+// waiting while canary create new topic on kafka broker.
+func (c *ServiceManager) waitForCanarySetUp(){
+	log.Println("start waiting for canary setup")
+	timeout := time.After(10 * time.Second)
+	canaryIsReadyChannel := make(chan bool)
+
+	go func() {
+		configuration := sarama.NewConfig()
+		brokers := []string{c.KafkaBrokerAddress}
+
+		for ;;{
+			consumer, err := sarama.NewConsumer(brokers, configuration)
```

Comment:
same as already mentioned before use the admin client to do the check on the canary topic and its existence.